hal/network/ncp_client/sara/sara_ncp_client.cpp

```diff
@@ -2069,8 +2123,18 @@ int SaraNcpClient::enterDataMode() {
 
     // There is some kind of a bug in 02.19 R410 modem firmware in where it does not accept
     // any AT commands over the second muxed channel, unless we send something over channel 1 first.
-    if (ncpId() == PLATFORM_NCP_SARA_R410) {
-        CHECK(waitAtResponse(parser_, 5000));
+    //
+    // If R510 and CONNECTED already, in the case where we are in low power mode and the keepalives
+    // may be diabled/longer, just in case we are trying to resume a broken connection, let's timeout
+    // faster with an AT/OK check prior to the following AT+CGATT? call that takes 90s to timeout.
+    // This will also skip a follow up AT+CPWROFF call because ready_ will be set to false.
+    if ((ncpId() == PLATFORM_NCP_SARA_R410) ||
```

Comment:
@technobly Can we leave `waitAtResponse()` for R410 as it was before just to reduce the set of affected platforms? There is a reason why it is that way (see comment above).

Reply:
If we leave that, and the AT interface dies... it will take a lot longer for the system to recover.  As far as I can tell from the previous code, it's just sending an AT/OK check on channel 1.  So this does the same thing, but also sets parserError_ in case of a timeout.

Reply:
I'm more concerned with R410. There is a separate reason we issue multiple AT/AT/AT requests here, so we can leave that as-is and just have execCommand for R510, right?

Reply:
Ok, from the comment, I didn't get that it requires multiple AT/AT/AT before OK on channel 1.  I believe that would still happen here, but I'll leave it the way it was for now just to not disturb it.